File: app/src/main/java/com/pluscubed/velociraptor/api/raptor/RaptorService.java
package com.pluscubed.velociraptor.api.raptor;

import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Single;

public interface RaptorService {
    @Headers("Content-Type: application/json")
    @POST("verify/{id}")
    Single<VerificationResponse> verify(@Path("id") String id, @Body String data);

    @GET("tomtom")
    Single<RaptorResponse> getTomtom(@Header("Authorization") String authorization, @Query("id") String id,
                                     @Query("lat") String lat, @Query("lng") String lng, @Query("vehicle_heading") int heading);

    @GET("here")
    Single<RaptorResponse> getHere(@Header("Authorization") String authorization, @Query("id") String id,
                                   @Query("lat") String lat, @Query("lng") String lng, @Query("vehicle_heading") int heading);
}

File: app/src/main/java/com/pluscubed/velociraptor/api/raptor/RaptorService.kt
package com.pluscubed.velociraptor.api.raptor

import retrofit2.Call
import retrofit2.http.*

interface RaptorService {
    @Headers("Content-Type: application/json")
    @POST("verify/{id}")
    fun verify(
        @Path("id") id: String,
        @Body data: String
    )
            : Call<VerificationResponse?>

    @GET("tomtom")
    fun getTomtom(
        @Header("Authorization") authorization: String,
        @Query("id") id: String,
        @Query("lat") lat: String,
        @Query("lng") lng: String,
        @Query("vehicle_heading") heading: Int,
        @Query("units") units: String
    )
            : Call<RaptorResponse?>

    @GET("here")
    fun getHere(
        @Header("Authorization") authorization: String,
        @Query("id") id: String,
        @Query("lat") lat: String,
        @Query("lng") lng: String,
        @Query("vehicle_heading") heading: Int,
        @Query("units") units: String
    )
            : Call<RaptorResponse?>
}
